mmengine/visualization/vis_backend.py

```diff
@@ -610,3 +611,176 @@ def close(self):
+            self._mlflow.log_params(self._params)
+
+    @property  # type: ignore
+    @force_init_env
+    def experiment(self):
+        """Return MLFlow object."""
+        return self._mlflow
+
+    @force_init_env
+    def add_config(self, config: Config, **kwargs) -> None:
+        """Record the config to mlflow.
+
+        Args:
+            config (Config): The Config object
+        """
+        self.cfg = config
+        self._mlflow.log_params(self._flatten(self.cfg))
+
+    @force_init_env
+    def add_image(self,
+                  name: str,
+                  image: np.ndarray,
+                  step: int = 0,
+                  **kwargs) -> None:
+        """Record the image to mlflow.
+
+        Args:
+            name (str): The image identifier.
+            image (np.ndarray): The image to be saved. The format
+                should be RGB.
+            step (int): Global step value to record. Default to 0.
+        """
+        self._mlflow.log_image(image, name)
+
+    @force_init_env
+    def add_scalar(self,
+                   name: str,
+                   value: Union[int, float, torch.Tensor, np.ndarray],
+                   step: int = 0,
+                   **kwargs) -> None:
+        """Record the scalar data to mlflow.
+
+        Args:
+              name (str): The scalar identifier.
+              value (int, float, torch.Tensor, np.ndarray): Value to save.
+              step (int): Global step value to record. Default to 0.
+        """
+        self._mlflow.log_metric(name, value, step)
+
+    @force_init_env
+    def add_scalars(self,
+                    scalar_dict: dict,
+                    step: int = 0,
+                    file_path: Optional[str] = None,
+                    **kwargs) -> None:
+        """Record the scalar's data to mlflow.
+
+        Args:
+                scalar_dict (dict): Key-value pair storing the tag and
+                    corresponding values.
+                step (int): Global step value to record. Default to 0.
+                file_path (str, optional): Useless parameter. Just for
+                    interface unification. Default to None.
+        """
+        assert isinstance(scalar_dict, dict)
+        assert 'step' not in scalar_dict, 'Please set it directly ' \
+                                          'through the step parameter'
+        self._mlflow.log_metrics(scalar_dict, step)
+
+    def close(self) -> None:
+        """Close the mlflow."""
+        if self._log_artifact:
+            work_dir = self.cfg.work_dir
+
+            for file in os.listdir(work_dir):
+                if file.endswith('.pth'):
+                    self._mlflow.log_artifact(os.path.join(work_dir, file))
+                elif file == 'last_checkpoint':
+                    self._mlflow.log_artifact(osp.join(work_dir, file))
+                elif file.startswith(osp.basename(osp.normpath(work_dir))):
+                    self._mlflow.log_artifact(osp.join(work_dir, file))
```

Comment:
Why do we not upload the whole working directory `work_dir`?

Reply:
If work_dir is specified as shown in the picture, there is a problem that the past records are also transmitted. Do you have any good ideas?

![image](https://user-images.githubusercontent.com/66167254/215263794-96453644-2c45-4ce3-a178-55d092d9c73f.png)

Reply:
I think it works as expected because all of the files in the `work_dir` belongs to the same experiment.

Reply:
reflected.